src/x11/x.c

```diff
@@ -833,6 +890,8 @@ gboolean x_mainloop_fd_dispatch(GSource *source, GSourceFunc callback,
                 switch (ev.type) {
                 case Expose:
                         if (ev.xexpose.count == 0 && xctx.visible) {
+                                x_win_round_corners();
+
                                 x_win_draw();
```

Comment:
There are many more calls of x_win_draw(). One for example is in [`run()` of dunst.c](https://github.com/dunst-project/dunst/blob/2d7dc6fbc592bf7a764286d8d8d42fda7db7549b/src/dunst.c#L223).

For my understanding the rounded corners have to be called every time the window gets drawn. Why don't you put it into `x_win_draw` directly?

Reply:
I do it in the following commit. I first thought, that its only necessary on the expose event.